### PR TITLE
Windows set up permission button #261

### PR DESCRIPTION
**Pull request by ert78gb**
It is solve the #261 request. But at the end of the installation I got a warning maybe the driver not installed correctly. But install correctly.
Win 7 32 bit

### CODE REVIEW

File: electron/src/components/privilege-checker/privilege-checker.component.ts [hunk 1]
```diff
@@ -81,4 +90,32 @@ export class PrivilegeCheckerComponent {
...
+        };
+        /**
+         * The parameters:
+             - vid: vendor ID
+             - pid: product ID
+             - iface: interface ID
+             - usealldevices: if the device has installed USB driver than overwrite it
+             - noprompt: return at the end of the installation and not waiting for any user command
+         */
+        const command = `"${scriptPath}" --vid 0x1D50 --pid 0x6122 --iface 0 --usealldevices --noprompt`;
```
Comment (mondalaci) [inline electron/src/components/privilege-checker/privilege-checker.component.ts]: The VID and PID values are also used elsewhere in the codebase. Please extract them and reference them, so they won't be duplicated.

**Comment by mondalaci** [comment]
Any objections, @fjozsef?

**Comment by fjozsef** [comment]
For me it stays on the: 
"Cannot find your UHK
Please plug it in!" screen and doesn't do anything.

**Comment by mondalaci** [comment]
Your prototype has the old VID and PID @fjozsef, that's the reason why. We'll meet soon and I'll give you a new one.

**Comment by mondalaci** [comment]
I'm not sure if we should pursue this one guys, given that we plan to move from libusb to [node-hid](https://github.com/node-hid/node-hid) based on [HIDAPI](https://github.com/signal11/hidapi) which should work out of the box on Windows.

**Comment by ert78gb** [comment]
ready to PR.
I tested on mac and Win 32 bit

**Comment by ert78gb** [comment]
This PR close: #314, #262,

File: electron/src/components/privilege-checker/privilege-checker.component.ts [hunk 2]
```diff
@@ -23,7 +24,19 @@ import { UhkDeviceService } from '../../services/uhk-device.service';
...
 
-    constructor(private router: Router, private uhkDevice: UhkDeviceService) {
+    private rootDir: string;
+
+    constructor(private router: Router,
+                private uhkDevice: UhkDeviceService,
+                @Inject(LOG_SERVICE) private logService: ILogService) {
+        if (isDev) {
+            this.rootDir = path.resolve(path.join(remote.process.cwd(), remote.process.argv[1]), '..');
+        }
```
Comment (mondalaci) [inline electron/src/components/privilege-checker/privilege-checker.component.ts]: `}` should be in the same line as the `else` keyword according to our coding standards.

File: electron/src/components/privilege-checker/privilege-checker.component.ts [hunk 3]
```diff
@@ -82,4 +101,33 @@ export class PrivilegeCheckerComponent {
         return subject.asObservable();
     }
 
+    private setUpPermissionsOnWin(): Observable<void> {
+        const subject = new ReplaySubject<void>();
+        /**
```
Comment (mondalaci) [inline electron/src/components/privilege-checker/privilege-checker.component.ts]: Let's rather use a single line comment here, please.

File: electron/src/services/uhk-device-provider.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Provider } from '@angular/core';
+
+import { UhkDeviceService } from './uhk-device.service';
+import { UhkHidApiService } from './uhk-hid-api.service';
+
+export function uhkDeviceProvider(): Provider {
+    // HID API officially support MAC, WIN and linux x64 platform
+    // https://github.com/node-hid/node-hid#platform-support
+    if (process.platform === 'darwin' ||
```
Comment (mondalaci) [inline electron/src/services/uhk-device-provider.ts]: Let's rather use hidapi exclusively. It should support every platform we need. This way, we'll be able to test hidapi on every platform and ditch libusb eventually if everything works.

https://github.com/node-hid/node-hid#platform-support : 

> ia32, ARM, MIPSel and PPC64 platforms are known to work but are not currently part of our test or build matrix. ARM v4 and v5 was dropped from Node.js after Node v0.10.

File: electron/src/components/privilege-checker/privilege-checker.component.ts [hunk 5]
```diff
@@ -50,14 +63,19 @@ export class PrivilegeCheckerComponent {
             case 'linux':
                 permissionSetter = this.setUpPermissionsOnLinux();
                 break;
+            case 'win32':
```
Comment (mondalaci) [inline electron/src/components/privilege-checker/privilege-checker.component.ts]: Please comment out this branch, and add a comment explaining that hidapi shouldn't need privilege escalation on Windows.

**Comment by mondalaci** [comment]
@fjozsef Any further objections?

File: electron/src/services/uhk-device-provider.ts (hunk 4 above)
Comment (ert78gb) [inline electron/src/services/uhk-device-provider.ts]: Should I do anything until we will finish the test on all platform?
Currently the app use UhkHidApiService on all platform.
If we drop UhkLibUsbApiService from the code, then I like to remove the uhkDeviceProvider and use directly UhkHidApiService every where.

File: electron/src/components/privilege-checker/privilege-checker.component.ts (hunk 5 above)
Comment (ert78gb) [inline electron/src/components/privilege-checker/privilege-checker.component.ts]: done

File: electron/src/services/uhk-device-provider.ts (hunk 4 above)
Comment (mondalaci) [inline electron/src/services/uhk-device-provider.ts]: Let's not drop `UhkHidApiService` for the time being, and do this refactoring after a fair amount of testing in a separate PR.

**Pull request merged**